pints/_optimisers/_cmaes_bare.py

```diff
@@ -0,0 +1,379 @@
+        self._zs = np.array([np.random.normal(0, 1, self._n_parameters)
+                             for i in range(self._population_size)])
+
+        # Centered samples: centered at zero, with rotation and scaling
+        self._ys = np.array([self._R.dot(self._S).dot(z) for z in self._zs])
+
+        # Samples from N(mu, eta**2 * C)
+        self._xs = np.array([self._mu + self._eta * y for y in self._ys])
+
+        # Apply boundaries; creating safe points for evaluation
+        # Rectangular boundaries? Then perform boundary transform
+        if self._boundary_transform is not None:
+            self._xs = self._boundary_transform(self._xs)
+
+        # Manual boundaries? Then pass only xs that are within bounds
+        if self._manual_boundaries:
+            self._user_ids = np.nonzero(
+                [self._boundaries.check(x) for x in self._xs])
+            self._user_xs = self._xs[self._user_ids]
+            if len(self._user_xs) == 0:     # pragma: no cover
+                self._logger.warning('All points requested by CMA-ES are'
+                                     ' outside the boundaries.')
+        else:
+            self._user_xs = self._xs
+
+        # Set as read-only and return
+        self._user_xs.setflags(write=False)
+        return self._user_xs
+
+    def cov(self, decomposed=False):
+        """
+        Returns the current covariance matrix ``C`` of the proposal
+        distribution.
+
+        If the optional argument ``decomposed`` is set to ``True``, a tuple
+        ``(R, S)`` will be returned such that ``R`` contains the eigenvectors
+        of ``C`` while ``S`` is a diagonal matrix containing the squares of the
+        eigenvalues of ``C``, such that ``C = R S S R.T``.
+        """
+        if decomposed:
+            return self._R, self._S
+        else:
+            return np.copy(self._C)
+
+    def fbest(self):
+        """ See :meth:`Optimiser.fbest()`. """
+        return self._fbest
+
+    def mean(self):
+        """
+        Returns the current mean of the proposal distribution.
+        """
+        return np.copy(self._mu)
+
+    def _initialise(self):
+        """
+        Initialises the optimiser for the first iteration.
+        """
+        assert (not self._running)
+
+        # Create boundary transform, or use manual boundary checking
+        self._manual_boundaries = False
+        self._boundary_transform = None
+        if isinstance(self._boundaries, pints.RectangularBoundaries):
+            self._boundary_transform = pints.TriangleWaveTransform(
+                self._boundaries)
+        elif self._boundaries is not None:
+            self._manual_boundaries = True
+
+        # Parent generation population size
+        # The parameter parent_pop_size is the mu in the papers. It represents
+        # the size of a parent population used to update our paramters.
+        # Not sure if the limitation to dim is good
```

Comment:
This comment is not clear to me. What is `dim` and where is it used to limit the `parent_pop_size`

Reply:
In line 176 it seems as if the `parent_pop_size` is simply always the smaller half of `population_size`